bzt/modules/apiritif/generator.py

```diff
@@ -653,6 +661,50 @@ def _gen_action(self, action_config):
 
         return [ast.Expr(element) for element in action_elements]
 
+    def _gen_answer_dialog_mngr(self, type, value):
+        if not type in ['prompt', 'confirm']:
+            raise TaurusConfigError("answerDialog type must be one of the following: 'prompt' or 'confirm'")
+        if type == 'confirm' and str(value).lower() not in ['#ok', '#cancel']:
+            raise TaurusConfigError("answerDialog of type confirm must have value either '#Ok' or '#Cancel'")
+        self.selenium_extras.add("DialogsManager")
+        dlg_method = "self.dlg_mng.answer_on_next_prompt" if type == 'prompt' else "self.dlg_mng.set_next_confirm_state"
+        return [ast_call(func=ast_attr(dlg_method), args=[ast.Str(value)])]
+
+    def _gen_assert_dialog_mngr(self, type, value):
+        if not type in ['alert', 'prompt', 'confirm']:
+            raise TaurusConfigError("assertDialog type must be one of the following: 'alert', 'prompt' or 'confirm'")
+        elements = []
+        self.selenium_extras.add("DialogsManager")
+        dlg_method = "self.dlg_mng.get_next_" + type
+
+        elements.append(ast.Assign(targets=[ast.Name(id='dialog', ctx=ast.Store())],
+                       value=ast_call(
+                            func=ast_attr(dlg_method))))
+        elements.append(ast_call(
+                func=ast_attr("self.assertIsNotNone"),
+                args=[ast.Name(id='dialog'), ast.Str("No dialog of type %s appeared" % type)]))
+        elements.append(ast_call(
+                func=ast_attr("self.assertEqual"),
+                args=[ast.Name(id='dialog'), ast.Str(value), ast.Str("Dialog message didn't match")]))
+
+        return elements
+
+    @staticmethod
+    def _gen_replace_dialogs():
```

Comment:
imho would be better to move try/catch into library (selenium_extras) and don't repeat it in generated code.
Could you also explain why this method may raise exception and why should we ignore it?

Reply:
The reason is that DialogsManager is only used when one of the new dialogs actions is present in the script. We probably don't want to replace the dialogs by the JS stubs in case there are no new dialogs method present as then the previous "alert" action will not work. 
I have made some changes as I agree that catching the exception is not so nice. Ignoring the exception just meant that we didn't instantiate the DialogsManager and that we are not replacing the dialogs... I hope that the new changes will make it more understandable...

Reply:
Sorry for vague explanation, I expect to see try/except in selenium_extras (not in user script) and I didn't ask to remove it at all. As I understood, it allows us to use DialogManger only if needed. (dialogs actions are present)